code/datums/diseases/mob_procs.dm

```diff
@@ -7,6 +7,10 @@
 
 // This proc has some procs that should be extracted from it. I believe we can develop some helper procs from it - Rockdtben
 /mob/proc/contract_disease(datum/disease/virus, skip_this = 0, force_species_check=1, spread_type = -5)
+	if(istype(virus, /datum/disease/black_goo) && (SSticker.mode && !MODE_HAS_TOGGLEABLE_FLAG(MODE_ALLOW_ZOMBIES)))
```

Comment:
Forest seriously I swear.... OOP exists for a reason. At least do it 2018 style and define a handler proc on the disease, don't strap istypes based chunks in the middle of stuff like it's 2012

Reply:
```
if(virus.can_infect(src))
```
or something i don't know

Reply:
But where's the fun in that. If I did everything perfectly first time around you wouldn't have so much fun reviewing it! :HAH:

Reply:
Mean :sob:

Reply:
Will do this when/if I have time tomorrow